services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/client/pulsar/DataPlaneClientImpl.java

```diff
@@ -102,22 +95,32 @@ private void createGoalState(MulticastGoalState multicastGoalState) throws Excep
 
             multicastGoalState.setNextTopics(groupTopics);
 
-            Producer<MulticastGoalStateByte> producer = pulsarClient
-                    .newProducer(JSONSchema.of(MulticastGoalStateByte.class))
-                    .topic(multicastTopic)
-                    .enableBatching(false)
-                    .create();
-
-            producer.send(multicastGoalState.getMulticastGoalStateByte());
+            try {
+                Producer<MulticastGoalStateByte> producer = pulsarClient
+                        .newProducer(JSONSchema.of(MulticastGoalStateByte.class))
+                        .topic(multicastTopic)
+                        .enableBatching(false)
+                        .create();
+
+                producer.send(multicastGoalState.getMulticastGoalStateByte());
+            } catch (Exception e) {
+                LOG.error("Send multicastGoalState to topic:{} failed: ", multicastTopic, e);
+                failedHosts.addAll(multicastGoalState.getHostIps());
+                continue;
+            }
 
             LOG.info("Send multicastGoalState to topic:{} success, " +
                             "groupTopics: {}, unicastGoalStates: {}",
                     multicastTopic, groupTopics, multicastGoalState);
         }
+
+        return failedHosts;
     }
 
     @Override
-    public List<Map<String, List<Goalstateprovisioner.GoalStateOperationReply.GoalStateOperationStatus>>> createGoalStates(List<UnicastGoalState> unicastGoalStates) throws Exception {
+    public List<String> createGoalStates(List<UnicastGoalState> unicastGoalStates) throws Exception {
+        List<String> failedHosts = new ArrayList<>();
+
         for (UnicastGoalState unicastGoalState: unicastGoalStates) {
```

Comment:
In the case of large number of neighbors, this for-loop could slow down potentially. See if we can optimize this part. @chenpiaoping